documentation/src/main/asciidoc/topics/upgrading.adoc

```diff
@@ -12,11 +12,98 @@ Use ```reindex``` instead:
+In previous versions `SingleFileStore` was the default for file-based cache stores.
+
+Any `file-store` configuration is automatically migrated to a `SoftIndexFileStore` after upgrading to Infinispan 13.
+
+=== Schema changes
+
+The following changes apply to the Infinispan 13 configuration schemas:
+
+* The `urn:infinispan:config:store:soft-index` namespace is no longer available.
+* The `file-store` element in the `urn:infinispan:config` namespace now defaults to soft-index file stores.
+* The `single-file-store` element is added to the `urn:infinispan:config` namespace if you want to configure a single file store, which is deprecated.
+
+=== Migrating Soft-Index File Stores
+
+==== Declarative
+
+If you are upgrading to Infinispan 13 from a previous version and your caches include any configuration with the `soft-index-file-store` element, you should convert that configuration to use the `file-store` element instead.
+
+.Infinispan 12.1 and earlier
+[source,xml,options="nowrap",subs=attributes+,role="primary"]
+----
+<persistence>
+  <soft-index-file-store xmlns="urn:infinispan:config:soft-index:12.1">
+    <index path="testCache/index" />
+    <data path="testCache/data" />
+  </soft-index-file-store>
+</persistence>
+----
+
+.Infinispan 13 and later
+[source,xml,options="nowrap",subs=attributes+,role="secondary"]
+----
+<persistence>
+  <file-store>
+    <index path="testCache/index" />
+    <data path="testCache/data" />
+  </file-store>
+</persistence>
+----
+
+==== Programmatic
+
+.Infinispan 12.1 and earlier
+[source,xml,options="nowrap",subs=attributes+,role="primary"]
+----
+ConfigurationBuilder b = new ConfigurationBuilder();
+b.persistence()
+    .addStore(SoftIndexFileStoreConfigurationBuilder.class)
+        .indexLocation("testCache/index");
+        .dataLocation("testCache/data")
+----
+
+.Infinispan 13 and later
+[source,xml,options="nowrap",subs=attributes+,role="secondary"]
+----
+ConfigurationBuilder b = new ConfigurationBuilder();
+b.persistence()
+    .addSoftIndexFileStore()
+        .indexLocation("testCache/index")
+        .dataLocation("testCache/data");
+----
+
+=== Migrating Single File Stores
+
+To use `SingleFileStore` as a file-based cache store with Infinispan 13 or later, do one of the following:
+
+.Declarative
+[source,xml,options="nowrap",subs=attributes+,role="primary"]
+----
+<persistence passivation="false">
+  <single-file-store shared="false"
+                     preload="true"
+                     fetch-state="true"
+                     read-only="false"/>
+</persistence>
+----
+
+.Programmatic
+[source,java,options="nowrap",subs=attributes+,role="secondary"]
+----
+ConfigurationBuilder b = new ConfigurationBuilder();
+b.persistence()
+    .addSingleFileStore()
+    .maxEntries(5000);
+----
+
+However, Infinispan does not recommend using `SingleFileStore` because it does not offer the same performance as `SoftIndexFileStore`.
```

Comment:
Actually the recommendation is more about scalability since the memory requirements for single file store is much higher than soft index.